src/vos/vos_obj.c

```diff
@@ -930,8 +931,11 @@ singv_iter_fetch(struct vos_obj_iter *oiter, vos_iter_entry_t *it_entry,
 	it_entry->ie_vis_flags = VOS_VIS_FLAG_VISIBLE;
 	it_entry->ie_epoch	 = key.sk_epoch;
 	it_entry->ie_minor_epc	 = key.sk_minor_epc;
-	if (it_entry->ie_epoch <= oiter->it_punched)
-		it_entry->ie_vis_flags = VOS_VIS_FLAG_COVERED;
+	if (it_entry->ie_epoch <= oiter->it_punched.pr_epc) {
+		if (it_entry->ie_epoch < oiter->it_punched.pr_epc ||
+		    it_entry->ie_minor_epc <= oiter->it_punched.pr_minor_epc)
```

Comment:
There are four different code snippets for this visibility check (comparing filter's epoch and minor epoch with entry's) in this patch, it's better to unify them and make it into an inline function.